File: backend/api/src/altasai/evaluation/evaluateIntent.ts
import samples from '../datasets/intent-samples.json';
import { classifyIntent } from '../nlp/intentClassifier';
import { evaluateClassification, makeResult } from './metrics';

export const evaluateIntent = () => {
  const expected = samples.map((sample) => sample.expectedIntent);
  const actual = samples.map((sample) => classifyIntent(sample.text).label);
  const metrics = evaluateClassification(expected, actual);
  return makeResult('intent_classification', metrics.accuracy, 0.8, metrics);
};

File: backend/ml-service/app/evaluation/evaluate_intent.py
from app.nlp.intent_classifier import load_intent_samples, predict_intent
from app.evaluation.metrics import classification_metrics, pass_result


def evaluate() -> dict:
    samples = load_intent_samples()
    expected = [sample["label"] for sample in samples]
    actual = [predict_intent(sample["text"])["label"] for sample in samples]
    metrics = classification_metrics(expected, actual)
    return pass_result("intent_classifier", metrics["accuracy"], 0.85, metrics)
